src/plugins/marks/index.tsx

```diff
@@ -466,6 +530,46 @@ export class MarksPlugin {
         emit(...wrapped.unfold(token));
         }));
     });
+
+    // Handles up and down in autocomplete
+    this.api.registerHook('session', 'move-cursor-insert', async (struct, info) => {
+      const { action } = info;
+      const line = await that.document.getText(this.session.cursor.row);
+      const curMark = that.getMarkUnderCursor(line, this.session.cursor.col);
+      if (curMark === null) { return; };
+      const n = this.autocomplete_matches.length;
+      if (action === 'down') {
+        struct.preventDefault = true;
+        this.autocomplete_idx = ((this.autocomplete_idx % n) + n + 1) % n;
+      }
+      if (action === 'up') {
+        struct.preventDefault = true;
+        this.autocomplete_idx = ((this.autocomplete_idx % n) + n + n - 1) % n;
+      }
+    });
+
+    // Handles enter in autocomplete
+    this.api.registerHook('session', 'split-line', async (struct) => {
+      const line = await that.document.getText(this.session.cursor.row);
+      const curMark = that.getMarkUnderCursor(line, this.session.cursor.col);
+      if (curMark === null) { return; };
+      if (this.autocomplete_matches) {
+        struct.preventDefault = true;
+        // Set mark text
+        const matches = this.getMarkMatches(line);
+        const cursor = this.session.cursor;
+        const match = this.autocomplete_matches[this.autocomplete_idx];
+        await Promise.all(matches.map(async pos => {
+          if (cursor.col >= pos[0] && cursor.col <= pos[1]) {
+            const start = line[pos[0]] === '@' ? pos[0] + 1 : pos[0] + 2;
+            const end = line[pos[0]] === '@' ? pos[1] : pos[1] - 2;
+            const mutation = new ChangeChars(cursor.row, start, end - start, undefined, match.split(''));
+            await this.session.do(mutation);
+            cursor.col = start + match.length;
+          }
+        }));
+      }
+    });
```

Comment:
maybe there should also be a way to disable the autocomplete menu?  could get annoying if you actually want to move down a line

Reply:
I think escape to return to normal mode is pretty intuitive, and its only 1 extra keystroke at worst. cant really think of a better system atm.

Reply:
k, i think it's 2 keystrokes but seems fine mainly since this only happens when you're autocompleting a mark.  also people shouldn't use arrow keys to move in insert mode lol